wbooks/src/app/screens/Search/index.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react';
+import { SafeAreaView, FlatList, ListRenderItem } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { Book } from '@interfaces/Book';
+import actionCreators from '@redux/book/actions';
+import { AppState } from '@interfaces/Redux';
+
+import BookItem from '../BookList/components/BookItem';
+
+import styles from './styles';
+import NoItems from './components/NoItems';
+
+function Search() {
+  const dispatch = useDispatch();
+  const books = useSelector<AppState, Book[]>(state => state.bookReducer.books);
+  const searchCriteria = useSelector<AppState, string>(state => state.bookReducer.filterBooks) || '';
+
+  useEffect(() => {
+    dispatch(actionCreators.getBooks());
+  }, [dispatch]);
+
+  const keyExtractor = ({ id }: Book) => String(id);
+  const renderItem: ListRenderItem<Book> = ({ item }) => <BookItem item={item} />;
+
+  const filteredBooks =
+    searchCriteria.length < 1
+      ? []
+      : books.filter(book => {
+          const SEARCH = searchCriteria.toLowerCase();
+          const TITLE = book.title.toLowerCase();
+
+          return TITLE.includes(SEARCH);
+        });
```

Comment:
Do this inside the `useSelector`, it has more performance

Reply:
Thing is that if i do this inside useSelector, then it returns just the books (wich is awesome, but...) i also kinda need the searchCriteria string in order to use it on the <NoItems /> component.

Reply:
I tell you can do the filter inside the `useSelector`, nothing else:
```
  const searchCriteria = useSelector<AppState, string>(state => state.bookReducer.filterBooks) || '';
  const books = useSelector<AppState, Book[]>(state => state.bookReducer.books.filter(book => book.title.toLowerCase().includes(searchCriteria.toLowerCase())));
...
data={books}
...
```